### PR TITLE
Cms Kit - Content  Admin Implementation

### PR DESCRIPTION
**Pull request by enisn**
## Summary
This PR includes CmsKit Content admin implementation without UI.

### CODE REVIEW

File: modules/blogging/src/Volo.Blogging.EntityFrameworkCore/Volo/Blogging/Blogs/EfCoreBlogRepository.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public EfCoreBlogRepository(IDbContextProvider<IBloggingDbContext> dbContextProv
 
         public async Task<Blog> FindByShortNameAsync(string shortName)
         {
-            return await DbSet.FirstOrDefaultAsync(p => p.ShortName == shortName);
+            return await (await GetDbSetAsync()).FirstOrDefaultAsync(p => p.ShortName == shortName);
```
Comment (ilkayilknur) [inline modules/blogging/src/Volo.Blogging.EntityFrameworkCore/Volo/Blogging/Blogs/EfCoreBlogRepository.cs]: Why these changes included in this branch?

Comment (ilkayilknur) [reply modules/blogging/src/Volo.Blogging.EntityFrameworkCore/Volo/Blogging/Blogs/EfCoreBlogRepository.cs]: I fixed it.

File: modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Contents/ContentController.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    [Area("cms-kit")]
+    [Route("api/cms-kit-admin/contents")]
+    public class ContentController : CmsKitAdminController, IContentAdminAppService
+    {
+        public ContentController(IContentAdminAppService contentAdminAppService)
+        {
+            ContentAdminAppService = contentAdminAppService;
+        }
+
+        protected IContentAdminAppService ContentAdminAppService { get; }
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Contents/ContentController.cs]: We can move this to the top of the constructor.

File: modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        public async Task ShouldCreateAsync()
+        {
+            var created = await _service.CreateAsync(new ContentCreateDto
+            {
+                EntityId = "test_case_id",
+                EntityType = "test_case_type",
+                Value = "Some long content"
+            });
+
+            created.ShouldNotBeNull();
```
Comment (ilkayilknur) [inline modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs]: We can compare created object properties here and make sure that it contains all the values we've given.

File: modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            var result = await _service.GetAsync(_data.Content_1_Id);
+
+            result.ShouldNotBeNull();
+        }
+
+        [Fact]
+        public async Task ShouldCreateAsync()
+        {
+            var created = await _service.CreateAsync(new ContentCreateDto
```
Comment (ilkayilknur) [inline modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs:58]: We should add duplicate EntityType/EntityId creation test case.

File: modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            var content = await _service.GetAsync(_data.Content_1_Id);
+            content.ShouldNotBeNull();
+            content.Value.ShouldBe(newValue);
+        }
+
+        [Fact]
+        public async Task ShouldDeleteAsync()
+        {
+            await _service.DeleteAsync(_data.Content_2_Id);
+
```
Comment (ilkayilknur) [inline modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs:116]: What happens if a user wants to call the `DeleteAsync` method again? We should add a test method for it.

**Review by ilkayilknur — CHANGES_REQUESTED** [review]

File: modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Contents/ContentController.cs (hunk 2 above)
Comment (enisn) [reply modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Contents/ContentController.cs]: You can check Commit ce852b4f

File: modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs (hunk 4 above)
Comment (enisn) [reply modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs:58]: Can you check Commit 42eb25c3

File: modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs (hunk 5 above)
Comment (enisn) [reply modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs:116]: You can check Commit 42eb25c3

File: modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs (hunk 3 above)
Comment (enisn) [reply modules/cms-kit/test/Volo.CmsKit.Application.Tests/Contents/ContentAdminAppService_Tests.cs]: Can you check Commit 3a58a459 ?

**Review by ilkayilknur — APPROVED** [review]

File: modules/cms-kit/src/Volo.CmsKit.EntityFrameworkCore/Volo/CmsKit/EntityFrameworkCore/CmsKitDbContextModelCreatingExtensions.cs [hunk 6]
```diff
@@ -100,9 +100,11 @@ public static class CmsKitDbContextModelCreatingExtensions
             builder.Entity<Content>(b =>
             {
                 b.ToTable(options.TablePrefix + "Contents", options.Schema);
-                
+
                 b.ConfigureByConvention();
-                
+
+                b.HasQueryFilter(x => !x.IsDeleted); // TODO: Find out why default filter doesn't work.
```
Comment (cotur) [inline modules/cms-kit/src/Volo.CmsKit.EntityFrameworkCore/Volo/CmsKit/EntityFrameworkCore/CmsKitDbContextModelCreatingExtensions.cs]: Please remove this query filter.

File: modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Contents/MongoContentRepository.cs [hunk 7]
```diff
@@ -51,5 +53,15 @@ public Task DeleteAsync(string entityType, string entityId, Guid? tenantId = nul
                     x.TenantId == tenantId,
                 cancellationToken: GetCancellationToken(cancellationToken));
         }
+
+        public async Task<bool> ExistsAsync([NotNull] string entityType, [NotNull] string entityId, Guid? tenantId = null, CancellationToken cancellationToken = default)
+        {
+            return await (await GetMongoQueryableAsync()).AnyAsync(x =>
+                    !x.IsDeleted &&
```
Comment (cotur) [inline modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Contents/MongoContentRepository.cs]: Please remove isDeleted filter

**Review by cotur — CHANGES_REQUESTED** [review]

**Review by ilkayilknur — CHANGES_REQUESTED** [review]
Can you pull the `dev` branch and remove the workarounds for the soft delete issues?

File: modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Contents/ContentController.cs [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+using System;
+using System.Threading.Tasks;
+using Volo.Abp;
+using Volo.Abp.Application.Dtos;
+using Volo.CmsKit.Admin.Contents;
+using Volo.CmsKit.Permissions;
+
+namespace Volo.CmsKit.Admin.Contents
+{
+    [Authorize(CmsKitAdminPermissions.Contents.Default)]
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Contents/ContentController.cs]: Please add global feature check here

File: modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Contents/ContentController.cs [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    [RemoteService(Name = CmsKitCommonRemoteServiceConsts.RemoteServiceName)]
+    [Area("cms-kit")]
+    [Route("api/cms-kit-admin/contents")]
+    public class ContentController : CmsKitAdminController, IContentAdminAppService
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Contents/ContentController.cs]: Let's rename this to the `ContentAdminController`. We need to use the same convention.

**Review by ilkayilknur — CHANGES_REQUESTED** [review]

**Review by ilkayilknur — APPROVED** [review]

**Comment by enisn** [comment]
@cotur Can you re-review?

**Review by cotur — APPROVED** [review]
